core/lib/Thelia/Module/BaseModule.php

```diff
@@ -521,7 +521,7 @@ public function install(ConnectionInterface $con = null)
      * @param string $newVersion the new module version, as defined in the module.xml file
      * @param ConnectionInterface $con
      */
-    public function update($currentVersion, $newVersion, ConnectionInterface $con)
+    public function update($currentVersion, $newVersion, ConnectionInterface $con = null)
```

Comment:
By adding the default "null" to $con parameter, you're breaking the compatibility with exiting modules which uses the signature without the default 'null' value, which was valid until now.

Please remove this 'null' default.